hydromt/models/model.py

```diff
@@ -161,19 +137,6 @@ def region(self) -> ModelRegionComponent:
         """Return the model region component."""
         return self.get_component("region", ModelRegionComponent)
 
-    @_classproperty
-    def api(cls) -> Dict:
-        """Return all model components and their data types."""
-        _api = cls._API.copy()
-
-        # reversed is so that child attributes take priority
-        # this does mean that it becomes imporant in which order you
-        # inherit from your base classes.
-        for base_cls in reversed(cls.__mro__):
-            if hasattr(base_cls, "_API"):
-                _api.update(getattr(base_cls, "_API", {}))
-        return _api
-
```

Comment:
Not sure if it was in the API docs but as the plugins used it we should document. Maybe add a section in the migration with the properties of model that were deleted/updated etc for properties other than the components?